src/lib/Liblog/pbs_log.c

```diff
@@ -858,7 +901,13 @@ log_record(int eventtype, int objclass, int sev, const char *objname, const char
 	if ((text == NULL) || (objname == NULL))
 		return;
 
-	now = time(NULL);	/* get time for message */
+	if (gettimeofday(&tp, NULL) != -1) {
+		now = tp.tv_sec;
+
+	if (pbs_conf.pbs_log_highres_timestamp)
+		snprintf(microsec_buf, sizeof(microsec_buf), ".%06ld", (long)tp.tv_usec);
+	} else
+		now = time(NULL);
```

Comment:
you do error checking for gettimeofday() but not for time(), why? I feel like we should just bail if gettimeofday fails.